planning/behavior_path_planner/src/scene_module/avoidance/debug.cpp

```diff
@@ -82,23 +82,91 @@ MarkerArray createAvoidLineMarkerArray(
   const auto disappearing_color = tier4_autoware_utils::createMarkerColor(0.9, 0.5, 0.9, 0.6);
 
-  Marker marker = createDefaultMarker(
-    "map", rclcpp::Clock{RCL_ROS_TIME}.now(), ns, 0L, Marker::CUBE,
-    createMarkerScale(3.0, 1.5, 1.5), normal_color);
-  int32_t i{0};
+  const auto uuid_to_id = [](const unique_identifier_msgs::msg::UUID & object_id) {
+    int32_t ret = 0;
+
+    for (size_t i = 0; i < sizeof(int32_t) / sizeof(int8_t); ++i) {
+      ret <<= sizeof(int8_t);
+      ret |= object_id.uuid.at(i);
+    }
+
+    return ret;
+  };
+
   MarkerArray msg;
-  for (const auto & object : objects) {
-    marker.id = ++i;
-    marker.pose = object.object.kinematics.initial_pose_with_covariance.pose;
-    marker.scale = tier4_autoware_utils::createMarkerScale(3.0, 1.5, 1.5);
-    marker.color = std::fabs(object.lost_time) < 1e-2 ? normal_color : disappearing_color;
-    msg.markers.push_back(marker);
+
+  {
+    auto marker = createDefaultMarker(
+      "map", rclcpp::Clock{RCL_ROS_TIME}.now(), ns, 0L, Marker::CUBE,
+      createMarkerScale(3.0, 1.5, 1.5), normal_color);
+    for (const auto & object : objects) {
+      marker.id = uuid_to_id(object.object.object_id);
+      marker.pose = object.object.kinematics.initial_pose_with_covariance.pose;
+      marker.scale = tier4_autoware_utils::createMarkerScale(3.0, 1.5, 1.5);
+      marker.color = std::fabs(object.lost_time) < 1e-2 ? normal_color : disappearing_color;
+      msg.markers.push_back(marker);
+    }
+  }
+
+  {
+    for (const auto & object : objects) {
+      const auto pos = object.object.kinematics.initial_pose_with_covariance.pose.position;
+
+      {
+        auto marker = createDefaultMarker(
+          "map", rclcpp::Clock{RCL_ROS_TIME}.now(), ns + "_envelope_polygon", 0L,
+          Marker::LINE_STRIP, createMarkerScale(0.1, 0.0, 0.0),
+          createMarkerColor(1.0, 1.0, 1.0, 0.999));
+
+        for (const auto & p : object.envelope_poly.outer()) {
+          marker.points.push_back(createPoint(p.x(), p.y(), pos.z));
+        }
+
+        marker.points.push_back(marker.points.front());
+        marker.id = uuid_to_id(object.object.object_id);
+        msg.markers.push_back(marker);
+      }
+    }
+  }
+
+  return msg;
+}
+
+MarkerArray createOtherObjectsMarkerArray(
+  const behavior_path_planner::ObjectDataArray & objects, std::string && ns)
+{
+  const auto normal_color = tier4_autoware_utils::createMarkerColor(0.0, 1.0, 0.0, 0.8);
+
+  const auto uuid_to_id = [](const unique_identifier_msgs::msg::UUID & object_id) {
+    int32_t ret = 0;
+
+    for (size_t i = 0; i < sizeof(int32_t) / sizeof(int8_t); ++i) {
+      ret <<= sizeof(int8_t);
+      ret |= object_id.uuid.at(i);
+    }
+
+    return ret;
+  };
+
+  MarkerArray msg;
+
+  {
+    Marker marker = createDefaultMarker(
+      "map", rclcpp::Clock{RCL_ROS_TIME}.now(), ns, 0L, Marker::CUBE,
+      createMarkerScale(3.0, 1.5, 1.5), normal_color);
+    for (const auto & object : objects) {
```

Comment:
personally I would put `msg.markers.reserve(objects.size())` here, since we already know the objects' size. How do you feel about this??

Reply:
Yes, I'll fix it :+1:

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/2282/commits/653c5800b449bc1b79c92223c388074da5f96f54